cmd/launcher/launcher.go

```diff
@@ -179,15 +181,22 @@ func runLauncher(ctx context.Context, cancel func(), opts *launcher.Options) err
 		"build", versionInfo.Revision,
 	)
 
+	var controlRegistry *control.Registry
```

Comment:
why use a *control.Registry var here? since *control.Registry is embedded in the ControlService you could just use
```suggestion
	var controlRegistry *control.ControlService
```

I guess it ensures that anyone outside the block below can only access registry

Reply:
Yeah that was my thinking - clients need only the reference to the Registry outside of the block below. I don't feel strongly either way though, whatever folks feel is most readable I'm cool with.

Reply:
Opted to simplify and get rid of the `Registry` idea all together, we can always introduce an interface here if it starts feeling not encapsulated enough.